### DISCUSSION TITLE
How to use mantine component Interface for create storybook's controls?

### DISCUSSION THREAD

**Discussion by emewjin** [mantinedev/mantine#1520] [Q&A] [answered]
### How to auto generate storybook's controls by mantine component Interface?

Hi, I just find out mantine component's storybook on mantine repo : https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/components/Button/Button.story.tsx

I run storybook via `yarn storybook` on mantine project, but cannot found controls in storybook.  
So I install `"@storybook/addon-controls": "^6.4.20",` but also nothing controls showing.
![image](https://user-images.githubusercontent.com/76927618/170492985-a138a735-09cc-4a57-98b5-2c162f06562c.png)

I try to fix `Button.story.tsx` like below code, but also nothing controls showing.
```ts
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import { Button } from './Button';

export default {
  title: 'Component/Button',
  component: Button,
  parameters: {},
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args}>Button</Button>;

export const Basic = Template.bind({});
```
![image](https://user-images.githubusercontent.com/76927618/170492716-b53dc769-7210-49bf-a88e-902879056d81.png)

As far as i know, storybook catch component's interface automatically by `as ComponentMeta<T>;` and use them to auto-generate controls options.
If cannot auto generate it by interface, I should type every options like this 😱
```ts
export default {
  title: 'Component/Button',
  component: Button,
  argTypes: {
    size: {
      control: {
        type: 'select',
        options: MANTINE_SIZES,
      },
    },
    variant: {
      control: {
        type: 'select',
        options: [
          'filled',
          'outline',
          'light',
          'gradient',
          'white',
          'default',
          'subtle',
        ],
      },
    },
    color: {
      control: {
        type: 'select',
        options: MANTINE_COLORS,
      },
    },
  },
} as ComponentMeta<typeof Button>;
```
So I try to use `ButtonProps<'button'>` as T, but nothing solved, just got below error.
![image](https://user-images.githubusercontent.com/76927618/170495246-9ef69d29-f2e9-48ba-8c74-2a9edbf51351.png)

I know there is demo in mantine docs, **but what I want is Storybook's controls addons pannel like below image.**
![image](https://user-images.githubusercontent.com/76927618/170493940-acba2f2b-30aa-470c-8fd2-8d1a40bcfec4.png)

I think mantine's developers also use storybook for develop UI components, testing UI. 
How did you create controls? please share 🙏🏻

**Comment by rtivital** [comment, marked as answer]
I do not use these controls, so I can only advise to follow storybook documentation or use `any` type
